llama_index/memory/types.py

```diff
@@ -0,0 +1,44 @@
+from abc import abstractmethod
+from pydantic import BaseModel
+from typing import List, Optional
+
+from llama_index.llms.base import ChatMessage, LLM
+
+
+class BaseMemory(BaseModel):
+    """Base class for all memory types."""
```

Comment:
not sure if these are the final abstractions, minor nit that it might be good to make a note in the docstring that this interface might change

Reply:
Definitely not final, will add that note